### PR TITLE
Fixed project naming for desired tag with dots.

### PR DESCRIPTION
**Pull request by mohammedzee1000**
When desired tag has dots, the project name also has the same
as it is appid-jobid-desired tag. This fails project creation
as openshift project names cannot have dots.
Added logic to relplace the same wherever project name was
being used.

### CODE REVIEW

**Comment by mohammedzee1000** [comment]
#dotests

File: server/run-test.sh [hunk 1]
```diff
@@ -35,8 +35,8 @@ docker pull ${FULL_FROM} || jumpto sendstatusmail
 
 _ "Re-tagging tested image (${FULL_FROM} -> ${FULL_TO})"
 docker tag ${FULL_FROM} ${FULL_TO} || jumpto sendstatusmail
-
-NS=${APPID}-${JOBID}-${DESIRED_TAG}
+printf "${APPID}${JOBID}${DESIRED_TAG}" | base64
```
Comment (navidshaikh) [inline server/run-test.sh]: why this repetition ?

Comment (mohammedzee1000) [inline server/run-test.sh]: ah that was from test but it should still be helpful to see the hash on screen

**Comment by mohammedzee1000** [comment]
Wow it passed

**Comment by mohammedzee1000** [comment]
@navidshaikh @dharmit if you guys can review this PR it would be good.

Once we start using hashes we might no longer have to worry about some of the other characters we check for as well in appid and jobid fields. The entire notion of replacing the characters came from the fact that they could not be part of openshift project names. 

Note : Openshift project names can be got by doing -

`$ printf "<jenkins-project-name>" | xxd -p `

File: beanstalk_worker/worker_start_build.py [hunk 2]
```diff
@@ -89,7 +90,10 @@ def start_build(job_details):
         appid = job_details['appid']
         jobid = job_details['jobid']
         desired_tag = job_details['desired_tag']
-        namespace = appid + "-" + jobid + "-" + desired_tag
+        pre_namespace = str(appid) + "-" + str(jobid) + "-" + str(desired_tag)
+        namespace = hexlify(pre_namespace)
+        debug_log("Openshift project namespace hexlified from {0} to {1} can be hash can be reproduced with xxd tool"
```
Comment (navidshaikh) [inline beanstalk_worker/worker_start_build.py]: *{1}, hash can be reproduced [..]

typo there

Comment (mohammedzee1000) [inline beanstalk_worker/worker_start_build.py]: Thanks will fix

**Comment by mohammedzee1000** [comment]
Ok appears to be working fine.

**Pull request merged**